src/node_file.cc

```diff
@@ -1095,6 +1099,46 @@ static void ExistsSync(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(err == 0);
 }
 
+bool FastExistsSync(v8::Local<v8::Object> recv,
+                    const FastOneByteString& string,
+                    // NOLINTNEXTLINE(runtime/references) This is V8 api.
+                    FastApiCallbackOptions& options) {
+  Environment* env = Environment::GetCurrent(recv->GetCreationContextChecked());
+
+  MaybeStackBuffer<char> path;
+
+  path.AllocateSufficientStorage(string.length + 1);
+  memcpy(path.out(), string.data, string.length);
+  path.SetLengthAndZeroTerminate(string.length);
```

Comment:
What's the reason for copying this data?

Reply:
I assumed it's for null termination, but I might be wrong.

Reply:
Yes, for null termination. Is this the right way to do it?

Reply:
AFAICT `auto input = std::string(string.data, string.length);` also handles null termination, and easier to read

Reply:
Doesn't `std::string` always incur a heap allocation? (I'll be the first to say that performance doesn't always matter, but I believe that's precisely why we use `MaybeStackBuffer` in such cases.)

Reply:
Yep, and I assumed `MaybeStackBuffer` also handles "heap- spilling" in case the string is big enough?